### ISSUE TITLE
Question about 'wordpress_test_cookie' declared in wp-includes/default-constants.php

### ISSUE THREAD

**Issue by stefanos82** [ClassicPress/ClassicPress#605]
Is it something we want to use as out test cookie name or is it something we could rename to 'classicpress_test_cookie'?

**Comment by nylen** [comment]
The first thing to check is the plugins/themes in the WP directory. This doesn't cover everything but it searches a large codebase that is part of our existing ecosystem.

This change would have an impact on some caching plugins (which decide to switch the cache or not, based on this cookie) and a few others: https://wpdirectory.net/search/01EFQJ353X112ET10M2MEBQ965

In my opinion this is not worth the backwards compatibility breakage, and our approach so far has been to leave the internal names of pre-existing things as "wp" and "wordpress" so we would also be breaking from that approach by making this change.

**Comment by stefanos82** [comment]
I see; I guess lots of things cannot change for the better due to the backwards compatibility need.

What are the plans for future version(s)?

Are we going to see some major changes in PHP use, for example modernizing the code by replacing legacy PHP syntax to something much more appealing?

I really don't like using `array()` and I always use its short syntax; I find it cleaner and easier to follow in nested use.

**Comment by ginsterbusch** [comment]
> I really don't like using `array()` and I always use its short syntax; I find it cleaner and easier to follow in nested use.

I guess thats part of a V3 package. Alas, its probably too much of an effort changing all that without breaking everything. Aside of the fact that it might increase the level of difficulty for newcomers, which has always been a strength (and yes, also a downfall, but that can be fought by education!) of WP and thus, also CP :)

cu, w0lf.

**Comment by stefanos82** [comment]
If it was announced that ClassicPress is in the process to upgrade to a newer, modern PHP syntax, I'm sure people would start using tools such as `php-cs-fixer` that does this for us, thus the rapid code improvement by community.

**Comment by stefanos82** [comment]
TLDR; I really want to help, but...let me try to explain as much as I can, even though it may sound like a rant.

For years I have been looking for a specific development model, something like sub-kanban tasks for sub-kanban tasks for kanban tasks, if you understand what I mean.

In my mind, by breaking down tasks or chores to their smallest parts possible makes it much more appealing, even for the uninitiated developer, let alone the professionally inexperienced one.

It would be really cool to have such mechanism available and say "oh man, I'm bored today; let's see whether there's a simple task in 'in process' kanban or not; I might learn something new today!".

It should feel like going for shopping to get your groceries, while crossing out the items you have just put in your cart.

"Done that, check; yep, that too; check; ooooh this is interesting! time to get some fun!"...that kind of mentality and feeling.

This is something I have been discussing for years with friends and acquaintances and haven't found anything alike it yet.

The majority of candidates or those who show genuine interest get overwhelmed by the amount of knowledge and expectation a project may have and they get exasperated by the outrageously huge prerequisite knowledge list.

I have been working with WordPress for more than a year now - on a personal level that is - and I can honestly say, before I learn something in depth, they either change it or add new content to it that transforms how it works or turn their direction elsewhere and leave you without alternative options.

That's more like a generalization about how WP works, but I honestly want to see projects like ClassicPress succeed based on their simplicity, stability, robustness, enterprise-y vibe, LTS support.

I can't keep going like this and constantly losing my sleep because a customer's website broke all of a sudden, completely out of the blue, because WordPress decided to release updates that didn't check it countless times for bugs and ooooh what do you know; your customer just lost a few dozen of orders, because that X bug decided to kick in **during the freaking midnight**!

I really want a stable system...am I asking too much?

If I can help in any way that is easy to do so, I would gladly do so...but if it's a Google "[6 Layers of Security](https://www.youtube.com/watch?v=kd33UVZhnAA)" procedure to fix a single line of code, forget it.

**Comment by nylen** [comment]
> I honestly want to see projects like ClassicPress succeed based on their simplicity, stability, robustness, enterprise-y vibe, LTS support

> I really want a stable system...am I asking too much?

This is what we want also. It sounds simple enough but in practice there really are a lot of things that need to be looked at in order to be able to make changes without breaking existing use cases.

Note that this set of goals is usually contradictory to the idea of refactoring "to make things prettier", especially if the refactoring is done piecemeal or not with full knowledge of what is being changed and what effects that may have downstream!

So, we're happy to have you, but you will have to find your own way through the changes you make instead of looking for "sub-kanban tasks for sub-kanban tasks for kanban tasks", because by the time you break things down to that level you might as well just do the work. I think you will find that this is true anywhere.

If you haven't seen it yet, please also take a look at our [review criteria for changes](https://github.com/ClassicPress/ClassicPress/blob/develop/.github/CONTRIBUTING.md#review-criteria), that will tell you pretty clearly what we (currently I) look at when evaluating changes.

**Comment by nylen** [comment]
Also, if you have questions about a task that we are looking to achieve, of course we'd be happy to help point you in the right direction. One example is adding the notice for site owners to upgrade their version of PHP: #438

This has an open PR but the PR doesn't have much in the way of explanation and some of the commits to backport WP changesets are malformed: #604

So a good next step for this particular change (as it says on the issue) is to look through the WP tickets and prepare a set of commits that we need to accept from WP. This should include everything that introduced and later modified their PHP version notification. The more information the better, such as links to WP tickets and context, so that others can follow and verify your train of thought.

**Comment by stefanos82** [comment]
So, ClassicPress will remain dependent on WordPress even with future versions..? :thinking: Because in my mind a fork is a project that takes a completely different direction than its original version; frankly, right now I'm confused...**a lot**! :confused:

My only guess is that the original intentions of ClassicPress' team was to have a WordPress LTS, but due to the copyright restriction of WordPress name, you had no other choice than come up with a different name :shrug:

**Comment by nylen** [comment]
Everyone seems to have a different opinion about what our stance should be regarding WP compatibility and newer features, but very few of these people seem to be willing to put in the work needed to advance their ideas.

In this case we clearly need site owners to understand the benefits of upgrading to the latest PHP version, and the risks of not doing so. WP has already done the work to make this happen, so why wouldn't we take advantage of that?

**Comment by stefanos82** [comment]
> Everyone seems to have a different opinion about what our stance should be regarding WP compatibility and newer features, but very few of these people seem to be willing to put in the work needed to advance their ideas.

Actually lots of these people want to help, but have no idea where to start, due to the bloated chaos WordPress is.

- Which file do I discard from WP, which one I don't?
- Do I need this X feature or not?
- Why we need proprietary mechanism embedded in WordPress.org that belongs to Automattic, such as Gravatar?
- Why is this X mechanism in WP when it could be as a plugin?

You see where this goes...
 
> In this case we clearly need site owners to understand the benefits of upgrading to the latest PHP version, and the risks of not doing so. WP has already done the work to make this happen, so why wouldn't we take advantage of that?

We cannot blame side owners for obvious reasons: they are not techies like us; they are end-users that paid somewhere to setup a website for them so they can do their job.

Last week I have had 2 separate calls from people who wanted me to help them with their mess, because they have setup their own WP website, even though they have nothing to do with technology, and after they broke things up badly, they wanted my expertise to help them.

Of course I rejected both for obvious reasons.

This led me to conclude that no matter what we do with WordPress, people will continue using obsolete, insecure versions of WordPress and PHP simply because they have no idea how to do an upgrade, let alone fix their code base; and if they do attempt such thing, they will get bitten by warnings and fatal errors from both CMS and language itself, thus the panic mode on their side.

It's so easy to download WP, extract its content on a hosting service, repeating the same procedure with a number of plugins, and then call yourself "a WordPress expert".

Imagine how I feel when I meet with people that claim they are experts in this field and I ask them questions about issues I have with specific plugin development features; 99 out of 100 times they say "eeeeeh you misunderstood me; I simply download or buy plugins to do my job and that should be it"!

I guess ClassicPress can change this at a very far future time when it would establish a steady audience and their majority will agree to proceed with backwards compatibility breakage, there's no other way around.

**Comment by nylen** [comment]
> Actually lots of these people want to help, but have no idea where to start,

The solution to this is to read, keep up with the project and ask questions. It will always be necessary to advance our own learning as developers and that is especially true when doing something as complicated as a full fork of WordPress.

> Why we need proprietary mechanism embedded in WordPress.org that belongs to Automattic, such as Gravatar?

This is likely the first of the questions in your latest comment which will see some definitive progress. In this case: https://petitions.classicpress.net/posts/8/disable-gr-avatar-by-default

The next question would be "how do we do this without breaking anything?" As for that: further investigations are welcome, but the sub-tasks won't (can't) be handed to anyone on a silver platter because we don't know what approach will work best until we start trying things.

We have discussed a couple of potential approaches to the general problem of core plugins during our [latest core development meeting](https://forums.classicpress.net/t/core-team-meeting-transcript-august-6-2020/2416).

As far as when that may happen.. the answer is "when someone does it". If that someone is me then it will probably be sometime in 2021 after launching our plugin directory.

I don't see what that has to do with PHP compatibility though which has a much simpler solution: nudge site owners to update to the latest version. Yes, this may cause issues in itself; so will not upgrading. We can present this information and what to do about it (and indeed we have a responsibility to do this so that people can keep their sites secured), that's about all.

**Comment by nylen** [comment]
Closing this, but continued discussion welcome.
